File: docs/reference/libtracker-sparql/examples/notifier-example.py
#!/usr/bin/python3

import gi, sys
from gi.repository import GLib, Gio, Tsparql

def callback(service, graph, events):
    for event in events:
        print('Event {0} on {1}\n'.format(
            event.get_event_type(), event.get_urn()))

try:
    connection = Tsparql.SparqlConnection.bus_new(
        'org.freedesktop.Tracker3.Miner.Files',
        None, None)

    notifier = connection.create_notifier()
    notifier.connect('events', callback)

    loop = GLib.MainLoop.new(None, False)
    loop.run()

    connection.close()

except Exception as e:
    print('Error: {0}'.format(e))
    sys.exit(-1)

File: docs/reference/libtracker-sparql/examples/notifier-example.js
#!/usr/bin/gjs

const { GLib, Gio, Tsparql } = imports.gi

try {
    let connection = Tsparql.SparqlConnection.bus_new(
        'org.freedesktop.Tracker3.Miner.Files',
        null, null);

    let notifier = connection.create_notifier();
    notifier.connect('events', (service, graph, events) => {
        for (let event in events)
            print (`Event ${event.get_event_type()} on ${event.get_urn()}`);
    });

    let loop = GLib.MainLoop.new(null, false);
    loop.run();

    connection.close();
} catch (e) {
    printerr(`Error: ${e.message}`)
}
